Add documentation for the fix_network script (#157)
OpenFreeEnergy/IndustryBenchmarks2024: docs/source/public/overview.rst

* If this is a non-redundant edge (meaning that removing this edge would lead to a disconnected graph):

>   * Add a new edge to the network. We provide a script that will automatically find a suitable new edge, more information can be found :ref:`here <fix_networks>`.

Reviewer: Good job I missed that one!

2. The extract results script mentioned :ref:`here <inspecting results>` and the cleanup script mentioned :ref:`here <post-simulation cleanup>` both include output of which folder(s) and ``json`` file(s) contain errors and can be removed prior to starting new jobs.

.. _fix_networks:

Fixing broken networks
----------------------

> If removing a reproducibly failing edge leads to a disconnected graph, new edges need to be added in order to fix the broken network.

Reviewer 2: How would they know that this is a broken network? I.e. number of ligands lower than expected, failure during the analysis scripts....

> The fixing of disconnected networks is carried out using the script provided under

Reviewer 2: ```suggestion
This can be achieved using the script provided under
```

`utils/fix_networks.py <https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/tree/main/industry_benchmarks/utils/fix_networks.py>`_.

> The script finds new edges to repair the disconnected network and produces the inputs necessary to run those additional edges.

Reviewer 2: How many new edges?

Note that the script will throw an error if at least one repeat completed successfully. In that case (non-reproducible failure) we recommend re-running the failed jobs (see :ref:`Handling failed edges <failed_edges>`

Here an example of how to run the script:

.. code-block:: bash

   python fix_networks.py --input_alchem_network_file alchemicalNetwork/alchemical_network.json --result_files results_*/*json --output_extra_transformations new_edges

> The script takes as inputs the ``AlchemicalNetwork`` from the original setup, the result ``.json`` files, and the folder name for storing the outputs.

Reviewer 2: Where should they expect to find this if they use the default instructions we gave them?

This command will create a folder (named ``new_edges`` as specified using the ``--output_extra_transformations`` flag) that contains a ``transformations`` folder with a separate .json file for the solvent and complex legs for every new edge connecting the previously broken network.

> The ``new_edges`` folder also contains a ``ligand_network.graphml`` file as well as an ``alchemical_network.json`` file containing the serialized version of the new ``LigandNetwork`` and ``AlchemicalNetwork``, respectively.

Reviewer 2: Clarify that this is only for the additional edges not the full new network?
